### DISCUSSION TITLE
Nature of Maps in Bytedeck

### DISCUSSION THREAD

**Discussion by mattjmoore** [bytedeck/bytedeck#1671] [Q&A]
Hi,

I am new to using Bytedeck so this is a very basic question.  I haven't started making anything in it yet, but my questions is about how the maps work.

From what I understand I should create quests and campaigns and then add those to courses later.  My question is about the Map in my deck.  Will the map show all the quests and campaigns that I have ever created, or will there be a separate map for each course that I create only showing the quests and campaigns that I add to that course?

From what I see so far playing around, a quest can only be part of one campaign (there was dropdown selection that only allowed me to choose one campaign at a time).  So do I have to make multiple copies of quests that I would want to be part of multiple courses?  For example, I am teaching a First Peoples 12 course and a Socials 11 and there are lessons I might want to use in both courses as the material overlaps to some degree.

If I am going to use Bytedeck for two courses with very different content, will the content from each course have to be visible on the map of the other course?  For example, I want to use Bytedeck to organize the material in a Math 10 course, but I don't want those quests to show up on the map for Socials 11 or First Peoples 12.

My overall question is just simply how do the quests and campaigns you organize into courses end up getting presented within the courses you later create?

Thanks

**Comment by tylerecouture** [comment]
Welcome Matt!  Thanks for the great questions.  You've prompted me to write an initial article in our Wiki to explain Maps, which when completed will be here: [Maps](https://github.com/bytedeck/bytedeck/wiki/Maps).

Although your deck can only have a single "main" map, i.e. the default map when you click the Map menu item.  You can have as many sub maps as you want. 

I teach five courses in my own deck, here how I set mine up (see the screenshot below)

1. Start with an overall intro quest that is short, easy, and automatically approved, it grants a badge just to give them a feel for how it all works
2. They then access the Orientation campaign, which they can do in any order.  These are skills commons to all courses, (how to use bytedeck, how to take screenshots or recordings, or submit their work, common classroom rules, etc).  
3. When they are all completed the student earns the "Bytedeck Proficiency" badge (that happens automatically).  I use this Proficiency badge as a prerequisite for all sorts of stuff later on.
4. Now they get access to the Introduction to their specific course.  In the maps, this links to a new map that is specific to their course.  You can see some of the map items are blue, they can be clicked to lead to sub-maps.  My intro quest for each course is basically their course outline.

![image](https://github.com/user-attachments/assets/2b1993f8-de38-4c63-90a2-6f16f5828764)

Although I don't do it, you could also set the specific "course" object as a prerequisite to the course intro quest, which means students will only see the quest for Intro for the course they are registered for, in their quest list (but they will still see them all on the common map).

"So do I have to make multiple copies of quests that I would want to be part of multiple courses?"

No, I would avoid this if you can!  If you have quests you want available in multiple courses, there are a few ways to do it:
* You can set an OR prerequisite, which means to access this quest, students need to complete THIS THING (course 1) or THAT THING (course 2), both will grant students access (you can do this for a single quest, or for the entire campaign)
* You can put them in a common map, that students can access based on their rank, or based on earning a badge, or some other prerequisite or combo of prerequisites.  For example, I have a digital literacy campaign that I want all my students to have access to, regardless of which course they are in, so I put a rank as a prerequisite, then when they reach that rank, they gain access to the Digital Literacy campaign (I use different ranks to spread out the common campaign content, so they don't get overwhelmed at the beginning =).
